pkg/services/baremetal/host/state_machine.go

```diff
@@ -249,19 +257,6 @@ func (hsm *hostStateMachine) handleImageInstalling() actionResult {
 	}
 
 	actResult := hsm.reconciler.actionImageInstalling()
-	if _, ok := actResult.(actionComplete); ok {
-		hsm.nextState = infrav1.StateProvisioning
-	}
-	return actResult
-}
-
-func (hsm *hostStateMachine) handleProvisioning() actionResult {
-	if hsm.provisioningCancelled() {
```

Comment:
this check might still be relevant. Should we really remove it?

Reply:
@janiskemper 

This is handled here:

```
func (hsm *hostStateMachine) handleEnsureProvisioned() actionResult {
	if hsm.provisioningCancelled() {
		hsm.nextState = infrav1.StateDeprovisioning
		return actionComplete{}
	}
```

and here:

```
func (hsm *hostStateMachine) handleImageInstalling() actionResult {
	if hsm.provisioningCancelled() {
		hsm.nextState = infrav1.StateDeprovisioning
		return actionComplete{}
	}
```

I think that is enough. Or am I missing something?

Reply:
maybe it just looked weird in the changes of the PR. but yes, this is what is necessary!

Reply:
ok for you?